frontend/src/lib/components/RadioSelect/index.tsx

```diff
@@ -9,23 +11,75 @@ export interface RadioSelectType {
 
 interface RadioSelectProps {
     options: RadioSelectType[]
-    selectedOption: null | string
-    onOptionChanged: (key: string | null) => void
+    selectedOption: null | string | string[]
+    onOptionChanged: (key: string | string[] | null) => void
+    multipleSelection?: boolean
+    focusSelection?: boolean // will hide other choices after making a selection
 }
 
-export function RadioSelect({ options, selectedOption, onOptionChanged }: RadioSelectProps): JSX.Element {
+export function RadioSelect({
+    options,
+    selectedOption,
+    onOptionChanged,
+    multipleSelection,
+    focusSelection,
+}: RadioSelectProps): JSX.Element {
+    const isSelected = (option: RadioSelectType): boolean => {
+        return multipleSelection && selectedOption
+            ? selectedOption?.includes(option.key)
+            : selectedOption === option.key
+    }
+
+    const handleClick = (option: RadioSelectType | null): void => {
+        if (!option) {
+            onOptionChanged(null)
+            return
+        }
+
+        if (multipleSelection) {
+            if (selectedOption instanceof Array) {
+                const _selectedOptions = selectedOption
```

Comment:
This needs to be a copy - we should not mutate params like this.

`const _selectedOptions = [...selectedOption]`